The objective is to create a first model to use as a baseline for subsequent models and optimizations

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('../data/curated/features_news_USD_pair_EURUSD_2007_2018.csv')

In [3]:
pd.options.display.max_columns = None

In [4]:
df.describe()

week  forecast_error_ratio         year      quarter  \
count  9667.000000           9667.000000  9667.000000  9667.000000   
mean     26.502845             -9.752338  2012.585704     2.485052   
std      14.895965            325.529724     3.405910     1.110890   
min       1.000000         -11400.000000  2007.000000     1.000000   
25%      13.000000            -11.430000  2010.000000     1.000000   
50%      26.000000              0.000000  2013.000000     2.000000   
75%      39.000000              7.825000  2016.000000     3.000000   
max      52.000000           7100.000000  2018.000000     4.000000   

             month      weekday  open_released  high_released  low_released  \
count  9667.000000  9667.000000    9667.000000    9667.000000   9667.000000   
mean      6.473984     2.341988   12828.111203   12832.481639  12823.708596   
std       3.416592     1.232760    1301.384660    1301.980591   1300.689311   
min       1.000000     0.000000   10395.000000   10397.000000  10388.000000   
25%       3.000000     1.000000   11654.000000   11659.000000  11652.000000   
50%       6.000000     2.000000   13033.000000   13038.000000  13030.000000   
75%       9.000000     3.000000   13695.000000   13701.000000  13690.500000   
max      12.000000     5.000000   15979.000000   15986.000000  15974.000000   

       close_released  high_5_after   low_5_after  close_5_after  \
count     9667.000000   9667.000000   9667.000000    9667.000000   
mean     12827.966380  12836.586014  12819.621806   12828.122272   
std       1301.446191   1302.047861   1300.568262    1301.325621   
min      10388.000000  10390.000000  10360.000000   10360.000000   
25%      11657.000000  11660.500000  11650.500000   11657.000000   
50%      13036.000000  13042.000000  13029.000000   13037.000000   
75%      13694.000000  13707.000000  13689.000000   13698.000000   
max      15976.000000  16016.000000  15966.000000   15989.000000   

       volatility_5_after  pips_agg_5_after  pips_candle_5_after  \
count         9667.000000       9667.000000          9667.000000   
mean            16.964208          0.155891             0.163029   
std             14.536439         14.733378            14.617713   
min              0.000000       -131.000000          -141.000000   
25%              8.000000         -5.000000            -5.000000   
50%             13.000000          0.000000             0.000000   
75%             21.000000          6.000000             6.000000   
max            169.000000        131.000000           131.000000   

       high_10_after  low_10_after  close_10_after  volatility_10_after  \
count    9667.000000   9667.000000     9667.000000          9667.000000   
mean    12834.231509  12821.747698    12827.944243            12.483811   
std      1302.127575   1300.177556     1301.212276             8.917978   
min     10369.000000  10356.000000    10362.000000             0.000000   
25%     11664.000000  11653.000000    11661.000000             7.000000   
50%     13043.000000  13031.000000    13037.000000            10.000000   
75%     13703.000000  13692.000000    13698.000000            16.000000   
max     16001.000000  15984.000000    15992.000000           122.000000   

       pips_agg_10_after  pips_candle_10_after  high_15_after  low_15_after  \
count        9667.000000           9667.000000    9667.000000   9667.000000   
mean           -0.022137             -0.172442   12833.610944  12822.388125   
std            17.444426              9.274754    1301.939304   1300.087749   
min          -173.000000            -70.000000   10362.000000  10340.000000   
25%            -7.000000             -4.000000   11664.000000  11655.000000   
50%             0.000000              0.000000   13041.000000  13033.000000   
75%             7.000000              4.000000   13704.000000  13693.000000   
max           193.000000             88.000000   15994.000000  15967.000000   

       close_15_after  volatility_15_after  pips_agg_15_a

Ups, Another unpleasant surprise.. :-(

Why some columns have nan? (e.g. high_30_after)


In [5]:
df[(df.impact == 'High') & (df.high_30_after.isnull())].head(5)

actual country             datetime forecast forecast_error impact  \
3612  59.8B     USD  2012-01-18 08:56:00    27.3B         better   High   
3613  17.9B     USD  2012-02-15 08:54:00    45.4B          worse   High   
3615  10.1B     USD  2012-04-16 08:56:00    40.7B          worse   High   
4843   69.6     USD  2008-02-15 09:55:00     76.0          worse   High   
4844   70.5     USD  2008-03-14 09:55:00     69.0         better   High   

                                new previous previous_error  week  \
3612        TIC Long-Term Purchases     8.3B         better     3   
3613        TIC Long-Term Purchases    61.3B       accurate     7   
3615        TIC Long-Term Purchases   102.4B       accurate    16   
4843  Prelim UoM Consumer Sentiment     78.4          worse     7   
4844  Prelim UoM Consumer Sentiment     70.8         better    11   

                   datetime_gmt  forecast_error_ratio  year  quarter  month  \
3612  2012-01-18 13:55:00+00:00                119.05  2012        1      1   
3613  2012-02-15 13:55:00+00:00                -60.57  2012        1      2   
3615  2012-04-16 12:55:00+00:00                -75.18  2012        2      4   
4843  2008-02-15 14:55:00+00:00                 -8.42  2008        1      2   
4844  2008-03-14 13:55:00+00:00                  2.17  2008        1      3   

      weekday  open_released  high_released  low_released  close_released  \
3612        2          12827          12827         12812           12815   
3613        2          13078          13088         13074           13087   
3615        0          13047          13052         13045           13046   
4843        4          14678          14679         14670           14677   
4844        4          15614          15632         15607           15622   

      high_5_after  low_5_after  close_5_after  volatility_5_after  \
3612         12816        12803          12804                  13   
3613         13090        13084          13090                   6   
3615         13048        13044          13046                   4   
4843         14687        14671          14685                  16   
4844         15648        15613          15641                  35   

     direction_candle_5_after direction_agg_5_after  pips_agg_5_after  \
3612                     down                  down               -11   
3613                       up                    up                 3   
3615                     down                  down                 0   
4843                       up                    up                 8   
4844                       up                    up                19   

      pips_candle_5_after  high_10_after  low_10_after  close_10_after  \
3612                  -11          12818         12802           12815   
3613                    3          13098         13088           13095   
3615                    0          13049         13041           13043   
4843                    9          14708         14680           14704   
4844                   14          15682         15644           15680   

      volatility_10_after direction_candle_10_after direction_agg_10_after  \
3612                   16                        up                   down   
3613                   10                        up                     up   
3615                    8                      down                   down   
4843                   28                        up                     up   
4844                   38                        up                     up   

      pips_agg_10_after  pips_candle_10_after  high_15_after  low_15_after  \
3612                  0                    11          12818         12804   
3613                  8                     5          13105         13094   
3615                 -3                    -2          13048         13041   
4843                 27                    18          14707         14693   
4844                 58              

Ok. I created my candles with a window size of 15 and 30 minutes, starting from 00:00. I.e.

00:00 - 00:15
00:15 - 00:30
00:30 - 00:45 
...

As some news are published in between those periods (e.g. 09:55), the join with the pairs dataframe does not work for them (09:55 + 15 min = 10:10, so there is no match with any candle).

A potential solution could be to create candles individually per each new, creating 15/30 min candles starting when each new was published, not starting by 00:00 for the entire dataset.

However, does it really worth the effort for the baseline model? Perhaps not, as remeber why we are creating this candles:

- 5 min candles:  We need fine granularity here, as the hypotesis is that market reaction right after the publication of the new is key for the models. 
There are not nan here, as we rounded the publication of the new to the closest 5min slot.

- 15 / 30 min candles: Before the new publication, this will be used to know whether the market was flat or was having a high volatility. Thus, we can apply the same criteria to round to the closest candle.

After the publication, this features will be used to know whether the new caused a positive or negative reaction in the market. Thus, I think we can apply the same criteria to round to the closest candle.


Therefore, let´s just round to the closest candle in the code for now.



Before rounding the df up, let´s review how many / which news are affected by this. 


In [6]:
df_temp = df[df.high_30_after.isnull()]
set(df_temp.new)

{'Capacity Utilization Rate',
 'Chicago PMI',
 'Crude Oil Inventories',
 'Empire State Manufacturing Index',
 'Existing Home Sales',
 'Federal Budget Balance',
 'Federal Funds Rate',
 'Final Services PMI',
 'HPI m/m',
 'IBD/TIPP Economic Optimism',
 'ISM Non-Manufacturing PMI',
 'Industrial Production m/m',
 'JOLTS Job Openings',
 'NAHB Housing Market Index',
 'NFIB Small Business Index',
 'Natural Gas Storage',
 'Personal Spending m/m',
 'Philly Fed Manufacturing Index',
 'Prelim UoM Consumer Sentiment',
 'Revised UoM Consumer Sentiment',
 'Richmond Manufacturing Index',
 'TIC Long-Term Purchases'}

Quite a lot considering all news, regardless its impact. Let´s see just for those with 'High' impact

In [7]:
df_temp = df[(df.impact == 'High') & (df.high_30_after.isnull())]
set(df_temp.new)

{'Existing Home Sales',
 'Federal Funds Rate',
 'ISM Non-Manufacturing PMI',
 'Prelim UoM Consumer Sentiment',
 'TIC Long-Term Purchases'}

In [8]:
df_temp = df[df.impact == 'High']
len(set(df_temp.new))

34

Much better picture, just 5 news out of 34

I´ve done the modification in the code, let´s review how it´s now

In [9]:
df = pd.read_csv('../data/curated/features_rounded_news_USD_pair_EURUSD_2007_2018.csv')

In [10]:
df.describe()

week  forecast_error_ratio         year      quarter  \
count  9667.000000           9667.000000  9667.000000  9667.000000   
mean     26.502845             -9.752338  2012.585704     2.485052   
std      14.895965            325.529724     3.405910     1.110890   
min       1.000000         -11400.000000  2007.000000     1.000000   
25%      13.000000            -11.430000  2010.000000     1.000000   
50%      26.000000              0.000000  2013.000000     2.000000   
75%      39.000000              7.825000  2016.000000     3.000000   
max      52.000000           7100.000000  2018.000000     4.000000   

             month      weekday  open_released  high_released  low_released  \
count  9667.000000  9667.000000    9667.000000    9667.000000   9667.000000   
mean      6.473984     2.341988   12828.111203   12832.481639  12823.708596   
std       3.416592     1.232760    1301.384660    1301.980591   1300.689311   
min       1.000000     0.000000   10395.000000   10397.000000  10388.000000   
25%       3.000000     1.000000   11654.000000   11659.000000  11652.000000   
50%       6.000000     2.000000   13033.000000   13038.000000  13030.000000   
75%       9.000000     3.000000   13695.000000   13701.000000  13690.500000   
max      12.000000     5.000000   15979.000000   15986.000000  15974.000000   

       close_released  high_0_5_after  low_0_5_after  close_0_5_after  \
count     9667.000000     9667.000000    9667.000000      9667.000000   
mean     12827.966380    12836.586014   12819.621806     12828.122272   
std       1301.446191     1302.047861    1300.568262      1301.325621   
min      10388.000000    10390.000000   10360.000000     10360.000000   
25%      11657.000000    11660.500000   11650.500000     11657.000000   
50%      13036.000000    13042.000000   13029.000000     13037.000000   
75%      13694.000000    13707.000000   13689.000000     13698.000000   
max      15976.000000    16016.000000   15966.000000     15989.000000   

       volatility_0_5_after  pips_agg_0_5_after  pips_candle_0_5_after  \
count           9667.000000         9667.000000            9667.000000   
mean              16.964208            0.155891               0.163029   
std               14.536439           14.733378              14.617713   
min                0.000000         -131.000000            -141.000000   
25%                8.000000           -5.000000              -5.000000   
50%               13.000000            0.000000               0.000000   
75%               21.000000            6.000000               6.000000   
max              169.000000          131.000000             131.000000   

       high_5_10_after  low_5_10_after  close_5_10_after  \
count      9667.000000     9667.000000       9667.000000   
mean      12834.231509    12821.747698      12827.944243   
std        1302.127575     1300.177556       1301.212276   
min       10369.000000    10356.000000      10362.000000   
25%       11664.000000    11653.000000      11661.000000   
50%       13043.000000    13031.000000      13037.000000   
75%       13703.000000    13692.000000      13698.000000   
max       16001.000000    15984.000000      15992.000000   

       volatility_5_10_after  pips_agg_5_10_after  pips_candle_5_10_after  \
count            9667.000000          9667.000000             9667.000000   
mean               12.483811            -0.022137               -0.172442   
std                 8.917978            17.444426                9.274754   
min                 0.000000          -173.000000              -70.000000   
25%                 7.000000            -7.000000               -4.000000   
50%                10.000000             0.000000                0.000000   
75%                16.000000             7.000000                4.000000   
max               122.000000           193.000000               88.000000   

       high_10_15_after  low_10_15_after  close_10_15_after  \
count       9667.000000      9667.000000        

In [11]:
set(df.count())

{9667}

Ok, now there are no nans

In [12]:
df.head(1)

actual country             datetime forecast forecast_error impact  \
0  12.3B     USD  2007-01-08 15:00:00     5.4B         better    Low   

                   new previous previous_error  week  \
0  Consumer Credit m/m    -1.3B       accurate     2   

                datetime_gmt  forecast_error_ratio  year  quarter  month  \
0  2007-01-08 20:00:00+00:00                127.78  2007        1      1   

   weekday  open_released  high_released  low_released  close_released  \
0        0          13018          13020         13016           13018   

   high_0_5_after  low_0_5_after  close_0_5_after  volatility_0_5_after  \
0           13022          13016            13019                     6   

  direction_candle_0_5_after direction_agg_0_5_after  pips_agg_0_5_after  \
0                       same                      up                   1   

   pips_candle_0_5_after  high_5_10_after  low_5_10_after  close_5_10_after  \
0                      0            13020           13016             13017   

   volatility_5_10_after direction_candle_5_10_after direction_agg_5_10_after  \
0                      4                        down                     down   

   pips_agg_5_10_after  pips_candle_5_10_after  high_10_15_after  \
0                   -1                      -2             13022   

   low_10_15_after  close_10_15_after  volatility_10_15_after  \
0            13013              13019                       9   

  direction_candle_10_15_after direction_agg_10_15_after  \
0                           up                        up   

   pips_agg_10_15_after  pips_candle_10_15_after  high_5_0_before  \
0                     1                        1            13020   

   low_5_0_before  close_5_0_before  volatility_5_0_before  \
0           13016             13018                      4   

  direction_candle_5_0_before  pips_candle_5_0_before  high_10_5_before  \
0                        same                       0             13022   

   low_10_5_before  close_10_5_before  volatility_10_5_before  \
0            13016              13018                       6   

  direction_candle_10_5_before  pips_candle_10_5_before  high_15_10_before  \
0                         same                        0              13020   

   low_15_10_before  close_15_10_before  volatility_15_10_before  \
0             13015               13018                        5   

  direction_candle_15_10_before  pips_candle_15_10_before  high_15_30_after  \
0                          down                        -1             13022   

   low_15_30_after  close_15_30_after  volatility_15_30_after  \
0            13015              13017                       7   

  direction_candle_15_30_after direction_agg_15_30_after  \
0                         down                      down   

   pips_agg_15_30_after  pips_candle_15_30_after  high_30_45_after  \
0                    -1                       -3             13021   

   low_30_45_after  close_30_45_after  volatility_30_45_after  \
0            13013              13017                       8   

  direction_candle_30_45_after direction_agg_30_45_after  \
0                           up                      down   

   pips_agg_30_45_after  pips_candle_30_45_after  high_45_60_after  \
0                    -1                        1             13021   

   low_45_60_after  close_45_60_after  volatility_45_60_after  \
0            13013              13019                       8   

  direction_candle_45_60_after direction_agg_45_60_after  \
0                         down                        up   

   pips_agg_45_60_after  pips_candle_45_60_after  high_30_15_before  \
0                     1                       -1              13022   

   low_30_15_before  close_30_15_before  volatility_30_15_before  \
0             13015               13018                        7   

  direction_candle_30_15_before  pips_candle_30_15_before  high_45_30_before  \
0                            u

Let´s do a check that everything is correct:

**5 min candles - AFTER:**

**08.01.2007 19:55:00.000,1.30180,1.30201,1.30159,1.30185,9259.3 <<-- Released - OK**  
08.01.2007 20:00:00.000,1.30186,1.30217,1.30158,1.30186,11694.0001

**08.01.2007 20:00:00.000,1.30186,1.30217,1.30158,1.30186,11694.0001  <<-- 5 min after - OK** 
08.01.2007 20:05:00.000,1.30191,1.30199,1.30159,1.30174,7981.6999000000005

**08.01.2007 20:05:00.000,1.30191,1.30199,1.30159,1.30174,7981.6999000000005 <<-- 10 min after - OK**
08.01.2007 20:10:00.000,1.30184,1.30221,1.30130,1.30188,7472.6999

**08.01.2007 20:10:00.000,1.30184,1.30221,1.30130,1.30188,7472.6999 <<-- 15 min after - OK** 
08.01.2007 20:15:00.000,1.30185,1.30209,1.30166,1.30200,4547.9


**5 min candles - BEFORE:**

08.01.2007 19:40:00.000,1.30198,1.30209,1.30169,1.30192,14232.9002

**08.01.2007 19:45:00.000,1.30190,1.30200,1.30149,1.30179,13996.6004 <<-- 15 min before - OK**

**08.01.2007 19:50:00.000,1.30176,1.30225,1.30159,1.30182,8935.4 <<-- 10 min before - OK **

**08.01.2007 19:55:00.000,1.30180,1.30201,1.30159,1.30185,9259.3 <<-- 5 min before - OK **

08.01.2007 20:00:00.000,1.30186,1.30217,1.30158,1.30186,11694.0001

In [13]:
df.head(1)

actual country             datetime forecast forecast_error impact  \
0  12.3B     USD  2007-01-08 15:00:00     5.4B         better    Low   

                   new previous previous_error  week  \
0  Consumer Credit m/m    -1.3B       accurate     2   

                datetime_gmt  forecast_error_ratio  year  quarter  month  \
0  2007-01-08 20:00:00+00:00                127.78  2007        1      1   

   weekday  open_released  high_released  low_released  close_released  \
0        0          13018          13020         13016           13018   

   high_0_5_after  low_0_5_after  close_0_5_after  volatility_0_5_after  \
0           13022          13016            13019                     6   

  direction_candle_0_5_after direction_agg_0_5_after  pips_agg_0_5_after  \
0                       same                      up                   1   

   pips_candle_0_5_after  high_5_10_after  low_5_10_after  close_5_10_after  \
0                      0            13020           13016             13017   

   volatility_5_10_after direction_candle_5_10_after direction_agg_5_10_after  \
0                      4                        down                     down   

   pips_agg_5_10_after  pips_candle_5_10_after  high_10_15_after  \
0                   -1                      -2             13022   

   low_10_15_after  close_10_15_after  volatility_10_15_after  \
0            13013              13019                       9   

  direction_candle_10_15_after direction_agg_10_15_after  \
0                           up                        up   

   pips_agg_10_15_after  pips_candle_10_15_after  high_5_0_before  \
0                     1                        1            13020   

   low_5_0_before  close_5_0_before  volatility_5_0_before  \
0           13016             13018                      4   

  direction_candle_5_0_before  pips_candle_5_0_before  high_10_5_before  \
0                        same                       0             13022   

   low_10_5_before  close_10_5_before  volatility_10_5_before  \
0            13016              13018                       6   

  direction_candle_10_5_before  pips_candle_10_5_before  high_15_10_before  \
0                         same                        0              13020   

   low_15_10_before  close_15_10_before  volatility_15_10_before  \
0             13015               13018                        5   

  direction_candle_15_10_before  pips_candle_15_10_before  high_15_30_after  \
0                          down                        -1             13022   

   low_15_30_after  close_15_30_after  volatility_15_30_after  \
0            13015              13017                       7   

  direction_candle_15_30_after direction_agg_15_30_after  \
0                         down                      down   

   pips_agg_15_30_after  pips_candle_15_30_after  high_30_45_after  \
0                    -1                       -3             13021   

   low_30_45_after  close_30_45_after  volatility_30_45_after  \
0            13013              13017                       8   

  direction_candle_30_45_after direction_agg_30_45_after  \
0                           up                      down   

   pips_agg_30_45_after  pips_candle_30_45_after  high_45_60_after  \
0                    -1                        1             13021   

   low_45_60_after  close_45_60_after  volatility_45_60_after  \
0            13013              13019                       8   

  direction_candle_45_60_after direction_agg_45_60_after  \
0                         down                        up   

   pips_agg_45_60_after  pips_candle_45_60_after  high_30_15_before  \
0                     1                       -1              13022   

   low_30_15_before  close_30_15_before  volatility_30_15_before  \
0             13015               13018                        7   

  direction_candle_30_15_before  pips_candle_30_15_before  high_45_30_before  \
0                            u


Let´s do a check that everything is correct:

15 min candles:

08.01.2007 20:15:00.000,1.30185,1.30209,1.30166,1.30200,4547.9

08.01.2007 20:20:00.000,1.30196,1.30220,1.30160,1.30178,11674.2999

08.01.2007 20:25:00.000,1.30178,1.30199,1.30154,1.30185,4770.8

08.01.2007 20:30:00.000,1.30179,1.30198,1.30153,1.30171,8375.3

** 15-30: High: 13022- Low: 13015 - Close: 12017 - OK **

08.01.2007 20:30:00.000,1.30179,1.30198,1.30153,1.30171,8375.3

08.01.2007 20:35:00.000,1.30165,1.30198,1.30148,1.30173,8657.400000000001

08.01.2007 20:40:00.000,1.30169,1.30207,1.30127,1.30184,8652

08.01.2007 20:45:00.000,1.30195,1.30210,1.30159,1.30172,7440.1

** 30 - 45 : High: 13021- Low: 13013 - Close: 13017 - OK **


In [14]:
df.head(1)

actual country             datetime forecast forecast_error impact  \
0  12.3B     USD  2007-01-08 15:00:00     5.4B         better    Low   

                   new previous previous_error  week  \
0  Consumer Credit m/m    -1.3B       accurate     2   

                datetime_gmt  forecast_error_ratio  year  quarter  month  \
0  2007-01-08 20:00:00+00:00                127.78  2007        1      1   

   weekday  open_released  high_released  low_released  close_released  \
0        0          13018          13020         13016           13018   

   high_0_5_after  low_0_5_after  close_0_5_after  volatility_0_5_after  \
0           13022          13016            13019                     6   

  direction_candle_0_5_after direction_agg_0_5_after  pips_agg_0_5_after  \
0                       same                      up                   1   

   pips_candle_0_5_after  high_5_10_after  low_5_10_after  close_5_10_after  \
0                      0            13020           13016             13017   

   volatility_5_10_after direction_candle_5_10_after direction_agg_5_10_after  \
0                      4                        down                     down   

   pips_agg_5_10_after  pips_candle_5_10_after  high_10_15_after  \
0                   -1                      -2             13022   

   low_10_15_after  close_10_15_after  volatility_10_15_after  \
0            13013              13019                       9   

  direction_candle_10_15_after direction_agg_10_15_after  \
0                           up                        up   

   pips_agg_10_15_after  pips_candle_10_15_after  high_5_0_before  \
0                     1                        1            13020   

   low_5_0_before  close_5_0_before  volatility_5_0_before  \
0           13016             13018                      4   

  direction_candle_5_0_before  pips_candle_5_0_before  high_10_5_before  \
0                        same                       0             13022   

   low_10_5_before  close_10_5_before  volatility_10_5_before  \
0            13016              13018                       6   

  direction_candle_10_5_before  pips_candle_10_5_before  high_15_10_before  \
0                         same                        0              13020   

   low_15_10_before  close_15_10_before  volatility_15_10_before  \
0             13015               13018                        5   

  direction_candle_15_10_before  pips_candle_15_10_before  high_15_30_after  \
0                          down                        -1             13022   

   low_15_30_after  close_15_30_after  volatility_15_30_after  \
0            13015              13017                       7   

  direction_candle_15_30_after direction_agg_15_30_after  \
0                         down                      down   

   pips_agg_15_30_after  pips_candle_15_30_after  high_30_45_after  \
0                    -1                       -3             13021   

   low_30_45_after  close_30_45_after  volatility_30_45_after  \
0            13013              13017                       8   

  direction_candle_30_45_after direction_agg_30_45_after  \
0                           up                      down   

   pips_agg_30_45_after  pips_candle_30_45_after  high_45_60_after  \
0                    -1                        1             13021   

   low_45_60_after  close_45_60_after  volatility_45_60_after  \
0            13013              13019                       8   

  direction_candle_45_60_after direction_agg_45_60_after  \
0                         down                        up   

   pips_agg_45_60_after  pips_candle_45_60_after  high_30_15_before  \
0                     1                       -1              13022   

   low_30_15_before  close_30_15_before  volatility_30_15_before  \
0             13015               13018                        7   

  direction_candle_30_15_before  pips_candle_30_15_before  high_45_30_before  \
0                            u


30 min candle:

08.01.2007 21:00:00.000,1.30145,1.30210,1.30126,1.30186,3453
08.01.2007 21:05:00.000,1.30171,1.30199,1.30136,1.30161,3559.8999999999996
08.01.2007 21:10:00.000,1.30156,1.30194,1.30120,1.30120,3800.3999999999996
08.01.2007 21:15:00.000,1.30180,1.30189,1.30143,1.30143,4587.700000000001
08.01.2007 21:20:00.000,1.30145,1.30215,1.30073,1.30200,3934.6
08.01.2007 21:25:00.000,1.30185,1.30207,1.30159,1.30191,3614.2
08.01.2007 21:30:00.000,1.30188,1.30215,1.30161,1.30165,3881.0000000000005

** 60 - 90 : High: 13021- Low: 13007 - Close: 13016 - OK **